src/Chirp.Infrastructure/CheepRepository.cs

```diff
@@ -166,4 +187,36 @@ public int GetCountOfAllCheepFromAuthor(string author)
         return db.Authors.Where(author => author.AuthorId == id).FirstOrDefault();
     }
     
-}
+    // Code directly from lecture
+    public async Task<Cheep> Create(CheepCreateDTO cheep)
+    {
+        //NullReferenceException is handled in the constructor - CheepRepository()
+        if (_validator == null) {
+            throw new NullReferenceException();
+        }
+        ValidationResult result = _validator.Validate(cheep);
+
+        if (!result.IsValid)
+        {
+            throw new ValidationException();
+        }
+        
+        var user = db.Authors.SingleAsync(u => u.Name == cheep.Author);
+
+        try
+        {
+             db.Add(new Cheep
+                {
+                    Author = user.Result,
+                    Text = cheep.Text,
+                    TimeStamp = DateTime.Now
+                });
+        }
+        catch (System.Exception)
+        {
+            throw;
+        }
```

Comment:
This try-catch is unecessary, since it throws the catched exception as the only action in the catch block, an therfor can be removed all together.

Reply:
This has been simplified in commit 3115815